src/model.rs

```diff
@@ -896,48 +902,63 @@ impl Collections {
             }
             fn find_stop_area_for<'a, F>(
                 collections: &'a Collections,
-                route_id: &'a str,
+                vehicle_journey_idxs: &IdxSet<VehicleJourney>,
                 select_stop_point_in_vj: F,
             ) -> Option<&'a StopArea>
             where
                 F: Fn(&VehicleJourney) -> Idx<StopPoint>,
             {
                 let stop_areas: Vec<Idx<StopArea>> =
-                    select_stop_areas(collections, route_id, select_stop_point_in_vj);
+                    select_stop_areas(collections, vehicle_journey_idxs, select_stop_point_in_vj);
                 let by_frequency: HashMap<Idx<StopArea>, usize> = group_by_frequencies(stop_areas);
                 let most_frequent_stop_areas = find_indexes_with_max_frequency(by_frequency);
                 let biggest_stop_areas =
                     find_biggest_stop_area(most_frequent_stop_areas, collections);
                 find_first_by_alphabetical_order(biggest_stop_areas)
             }
 
-            let origin_stop_area =
-                find_stop_area_for(collections, route_id, |vj| vj.stop_times[0].stop_point_idx);
-            let destination_stop_area = find_stop_area_for(collections, route_id, |vj| {
-                vj.stop_times[vj.stop_times.len() - 1].stop_point_idx
+            let vehicle_journey_idxs = routes_to_vehicle_journeys
+                .get_corresponding_forward(&IdxSet::from_iter(std::iter::once(route_idx)));
```

Comment:
We're using the relation to find fast the `vehicle_journey` corresponding to the `route`.